allensdk/test/api/cloud_cache/test_cache.py

```diff
@@ -746,3 +727,92 @@ def test_latest_manifest_warning(tmpdir, example_datasets_with_metadata):
     assert 'It is possible that some data files' in msg
     cmd = "S3CloudCache.load_manifest('project-x_manifest_v4.0.0.json')"
     assert cmd in msg
+
+
+@mock_s3
+def test_load_last_manifest(tmpdir, example_datasets_with_metadata):
+    """
+    Test that load_last_manifest works
+    """
+    bucket_name = 'load_lst_manifest_bucket'
+    metadatasets = example_datasets_with_metadata['metadata']
+    create_bucket(bucket_name,
+                  example_datasets_with_metadata['data'],
+                  metadatasets=metadatasets)
+
+    cache_dir = pathlib.Path(tmpdir) / 'load_last_cache'
+    cache = S3CloudCache(cache_dir, bucket_name, 'project-x')
+
+    # check that load_last_manifest in a new cache loads the
+    # latest manifest without emitting a warning
+    with pytest.warns(None) as warnings:
+        cache.load_last_manifest()
+    ct = 0
+    for w in warnings.list:
+        if w._category_name == 'OutdatedManifestWarning':
+            ct += 1
+    assert ct == 0
+    assert cache.current_manifest == 'project-x_manifest_v15.0.0.json'
+
+    cache.load_manifest('project-x_manifest_v7.0.0.json')
+
+    del cache
+
+    # check that load_last_manifest on an old cache emits the
+    # expected warning and loads the correct manifest
+    cache = S3CloudCache(cache_dir, bucket_name, 'project-x')
+    with pytest.warns(OutdatedManifestWarning) as warnings:
+        cache.load_last_manifest()
+    for w in warnings.list:
```

Comment:
I don't think you need to cycle through this list like this, I think the `pytest.warns()` context manager already asserts that there is this warning, and then you can use the `match=` to check that it is the warning message you wanted.

https://docs.pytest.org/en/latest/reference/reference.html#pytest.warns

Reply:
done